doc/en/user/source/services/csw/features.rst

```diff
@@ -44,11 +44,13 @@ The Internal Catalog Store currently supports two metadata schemes:
 Mapping Files
 -------------
 
-Mapping files are located in the ``csw`` directory inside the :ref:`datadir`. Each mapping file must have the exact name of the record type name combined with the ``.properties`` extension. For example:
+Mapping files are located in the ``csw`` directory inside the :ref:`datadir`. Assuming that each layer is mapped to a single record, each mapping file has the exact name of the record type name, combined with the ``.properties`` extension. For example:
 
 * Dublin Core mapping can be found in the file ``csw/Record.properties`` inside the data directory.
 * ISO Metadata mapping can be found in the file ``csw/MD_Metadata.properties`` inside the data directory (see :ref:`csw_iso` Community Module).
 
+There is also the possibility of having mapping each layer to multiple records. In this case, one would have multiple mapping files per type. For each type, each mapping file except maximum one must be given a name. The name must be encoded in the file name after the type name, followed by a dash, and before the extension. For instance: ``csw/Record-otherRecord.properties``, ``csw/MD_Metadata-thirdRecord.properties``. Geoserver will assume that mappings of different type but with the same name refer to the same record (in another format). (This is only relevant for GetRecords requests where outputSchema and typeNames do not match¸ which is unusual). The user is responsible for ensuring that identifiers are unique accross all mappings of the same record type.
```

Comment:
I read it twice, but understood nothing... "otherRecord" is the name of a layer here? I can't remember if we issue records also for the services, if we do, what would be the convention?
Please use the default data directory layer names to make things clearer in that case (e.g., topp:states, sf:roads), that would also help figuring out how namespaces are handled.

Bits that I'm having particular difficulty understanding:
- "each mapping file except maximum one must be given a name" -> another way to say that there can be a "Record.properties", but one can also go ahead and only have named files "Record-r1.properties", "Record-r2.properties" and so on? 
- "mapping each layer to multiple records"... so if we have "otherRecord" then all layers are mapped to two records?

I think this needs clearer explanation and a worked-out example of what would happen given a particular setup.

Reply:
Hello Niels, hello Andrea, 

here just some information of us (the GIS department of the city of Norderstedt) concerning - not the technical but the governmental - convention:

The SH-MIS (Metadata Information System of the federal state Schleswig-Holstein, Germany) wants us to offer a CSW record for every service. So we need a CSW for the WMS and a CSW for the WFS. Also they want us to offer a CSW for the "database" of the service/the place where the data is stored. 

So for almost every layer we host with geoserver (except for the raster, because it is only WMS), we have to offer three CSW records. This is the governmental convention we have to follow.

At the moment we offer round about 120 CSW-records and it will be more soon:
https://sh-mis.gdi-sh.de/catalog/?lang=de#/search?filter=&term=norderstedt&core=&sortAttribute=title_sort

We have got some pressure from SH-MIS to finish this with all needed CSW-records, because the SH-MIS is connected to the Open Data Portal of Schleswig-Holstein and the data of the SH-MIS can only be harvested form the Open Data Portal if the CSW's are complete.

So we are really looking forward if Niels work is merged in the csw- and the metadata-module. :)

Best regards,
Heiko Friedrich (Teamleader of the GIS-Department of Norderstedt)

Reply:
The use case is not relevant in the context of the review (thought it was interesting to read anyway). 
The documentation as written looks unclear and needs better explanation and examples, so that someone other than the implementor and the customer can make use of this feature.

Reply:
@aaime to answer your questions:

* "each mapping file except maximum one must be given a name" -> another way to say that there can be a "Record.properties", but one can also go ahead and only have named files "Record-r1.properties", "Record-r2.properties" and so on?  ----> yes, indeed

*    "mapping each layer to multiple records"... so if we have "otherRecord" then all layers are mapped to two records? ---> yes, indeed.

I will try to make the document clearer asap